### PR TITLE
1.1.5-3 - FP-2858: Removelog duplication for the second overlapping in requests

### PR DESCRIPTION
**Pull request by quirinpa**
- [FP-2858](https://movai.atlassian.net/browse/FP-2858): Removelog duplication for the second overlapping in requests

[FP-2858]: https://movai.atlassian.net/browse/FP-2858?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by diasdm — CHANGES_REQUESTED** [review]
This implementation seems to rely on backend sending all the logs for the given second and in the same order. But at least the first, we know its not truth :/

**Comment by quirinpa** [comment]
New implementation should overcome that issue.

File: package-lock.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
   "packages": {
     "": {
       "name": "@mov-ai/mov-fe-lib-react",
-      "version": "1.1.5-1",
+      "version": "1.1.5-2",
```
Comment (manuelnogueiramov) [inline package-lock.json:9]: Why are we changing the package-lock.json, but not the package.json?

Comment (quirinpa) [reply package-lock.json:9]: Because I did npm i. And that's the version number on package.json.

**Comment by diasdm** [comment]
Lets stick to the simpler implementation that does not change the order of logs already presented to the user, so:
* request 1: [a, b, d] 
* request 2: [a, c, d, e]
Should result in the following being presented to the user:
* render 1: [a, b, d] 
* render 2:  [a, b, d, c, e]

**Comment by quirinpa** [comment]
I think you mean "that does not insert logs between those the user already seen".
Because the order of the logs is currently preserved, but will not after we make that change.
That said, will do.

File: src/Components/Logs/Logs.js [hunk 2]
```diff
@@ -26,30 +26,71 @@ import i18n from "../../i18n/i18n";
...
+    }
+  }
+
+  // cut new logs up to z, cat with the deduped ones
+  // and the old logs up to i
+  return [].concat(
+    data.slice(0, z).map(log => transformLog(log)),
+    Object.values(map).sort(({ index: a }, { index: b }) => a - b),
+    oldLogs.slice(i),
+  );
```
Comment (manuelnogueiramov) [inline src/Components/Logs/Logs.js:90]: do we need the empty array here?

Would this work all the same?

```suggestion
  return data.slice(0, z).map(log => transformLog(log)).concat(
    Object.values(map).sort(({ index: a }, { index: b }) => a - b),
    oldLogs.slice(i),
  );
```

**Review by manuelnogueiramov — DISMISSED** [review]
Just a small non-blocking comment. 
Code still looks confusing, but the comments help :+1:

File: src/Components/Logs/Logs.js (hunk 2 above)
Comment (quirinpa) [reply src/Components/Logs/Logs.js:90]: I think it would still work, yes. Should I change it?

Comment (manuelnogueiramov) [reply src/Components/Logs/Logs.js:90]: Yeah. Test it before shipping tho! Good job.

**Comment by diasdm** [comment]
The last few commits:
- Simplify the log de-duplication algorithm
- Move the logs transformation outside the de-duplication
- Fix a bug where the current time was being used to fetch data (which in edge cases made a request for the second before the the timestamp of last log)

**Review by manuelnogueiramov — APPROVED** [review]
Good job. If @diasdm is happy, I'm happy :smile:

File: src/Components/Logs/Logs.js [hunk 3]
```diff
@@ -18,38 +18,79 @@ import {
   SIMPLE_LEVELS_LIST
 } from "./utils/Constants";
 import { findsUniqueKey } from "./utils/Utils";
-import useUpdateEffect from "./hooks/useUpdateEffect";
+// import useUpdateEffect from "./hooks/useUpdateEffect";
```
[the remark this answers is not in the document]
Comment (quirinpa) [reply src/Components/Logs/Logs.js]: Nope. Rmoved

File: src/Components/Logs/Logs.js [hunk 4]
```diff
@@ -389,13 +389,12 @@ const Logs = props => {
    * On change dates from filter
    */
   const onChangeDate = useCallback((newDate, keyToChange) => {
-    clearLogs();
```
Comment (manuelnogueiramov) [inline src/Components/Logs/Logs.js]: Are we sure this is no longer needed?

**Review by manuelnogueiramov — DISMISSED** [review]

File: src/Components/Logs/Logs.js (hunk 4 above)
Comment (quirinpa) [reply src/Components/Logs/Logs.js]: Yes. The useEffect thing does the job.

**Review by diasdm — APPROVED** [review]

**Pull request merged**